src/metabase/models/database.clj

```diff
@@ -12,7 +12,13 @@
 
 (i/defentity Database :metabase_database)
 
-(defn- pre-cascade-delete [{:keys [id] :as database}]
+(defn- post-select [{:keys [engine] :as database}]
+  (if-not engine database
+                 (assoc database :features (if-let [driver ((resolve 'metabase.driver/engine->driver) engine)]
+                                             ((resolve 'metabase.driver/features) driver)
```

Comment:
We're already passing driver features back as part of `window.MetabaseBootstrap`. Remember we had to add the code to load all the drivers when Metabase launches? I know it's a hassle but we don't need all this extra code so if we want to do it right we should revert it and use the stuff we already have.